### DISCUSSION TITLE
orari trasmissione fatture elettroniche a SDI

### DISCUSSION THREAD

**Discussion by Psicogest** [fattureincloud/api#339] [Q&A] [answered]
Buongiorno,

vorrei sapere da dove viene il "limite" secondo il quale viene impedita la trasmissione a SDI delle F.E. nel periodo (circa) 00:00 - 02:00 con messaggio di errore:
_"Stai provando a inviare una fattura con data odierna, tuttavia il Sistema di Interscambio considera le prime ore della notte del giorno corrente appartenenti al giorno precedente, pertanto scarterebbe il documento per \"data successiva alla data corrente\". Ti preghiamo di tentare l'invio domani mattina o dopo le 2 di notte o di cambiare la data del documento."_

Nei docs di SDI non c'è nulla del genere, e questa situazione genera non pochi problemi per la nostra integrazione...
(oltretutto si ottiene un errore anche inviando verso le 02:30 quindi oltre la fine del limite ...)

grazie

Ale

**Comment by valmoz** [comment]
Buongiorno @Psicogest 
il problema è legato ad un utilizzo errato da parte di SDI del GMT, che provoca un errore come indicato dal messaggio restituito. Per evitare l'insorgere di questa problematica, effettuiamo un controllo a monte in modo tale da bloccarlo a livello Fatture in Cloud, prima ancora dell'invio verso SDI.

Per quanto riguarda il problema delle 02.30, è effettivamente un bug lato nostro sul controllo in questione, che rifiuta anche i messaggi tra le ore 02:00 e le 02:59. Andremo a rilasciare una fix per questo bug, ma nel frattempo le suggerirei se possibile di iniziare i caricamenti a partire dalle ore 03:00 in modo tale da aggirare l'ostacolo.

La ringrazio, resto a disposizione per ulteriori chiarimenti.

Mauro

**Reply by Psicogest** [reply]
Buongiorno,

grazie per la rapida risposta, ma siete sicuri che questo problema di "fuso orario" di SDI ci sia ancora ? non è magari un problema risolto da loro da tempo ? Avevate segnalato a SDI questa cosa ?
Intanto allargo fino alle 03:00 il periodo di evitamento.

grazie

Ale

**Reply by valmoz** [reply]
Provvederò a segnalare ai colleghi che si occupano di questa parte di provare a indagare in merito. Le farò sapere nel caso in cui dovessero esserci delle novità a riguardo.

Restiamo a disposizione nel caso dovessero verificarsi altri dubbi o problemi. Per il momento chiuderò questa discussione, ma si senta libero di riaprirla in caso di necessità.

Grazie, le auguro una buona giornata
Mauro

**Reply by Psicogest** [reply]
Buongiorno,

vorrei invitarvi a riconsiderare (o meglio eliminare) questo controllo / limite, anche alla luce di quanto mi è stato detto nel seguente thread su Forum Italia Digitale:

https://forum.italia.it/t/orari-sdi/35669/7

grazie

Ale

**Reply by valmoz** [reply]
Buongiorno @Psicogest 
la pregherei la prossima volta di riaprire la discussione di conseguenza, in caso contrario potremmo perdercela e di conseguenza potrebbe non ricevere risposta.

Giro questa discussione ai colleghi che si occupano di questa parte dell'applicativo, non gestendolo io personalmente però non le posso assicurare niente.

In caso di aggiornamenti le risponderò su questo thread.

A disposizione
Mauro

**Reply by Psicogest** [reply]
Buongiorno,

ah scusate pensavo bastasse rispondere per ri-aprire il thread.

grazie

AdM

**Comment by Psicogest** [comment]
Buongiorno,
riapro questo thread perchè il problema si è ripresentato peggiorato, nel senso da stanotte sembra abbiate esteso questo inutile limite (vedi sopra) anticipandolo a già subito dopo la mezzanotte!
alle 00:05 di oggi (venerdiì 22/9) ho ottenuto questo errore dalle API di trasmissione a SDI:
`{"error":{"message":"Validation step 1","validation_result":{"xml_errors":["Stai provando a inviare una fattura con data odierna, tuttavia il Sistema di Interscambio considera le prime ore della notte del giorno corrente appartenenti al giorno precedente, pertanto scarterebbe il documento per \"data successiva alla data corrente\". Ti preghiamo di tentare l'invio domani mattina o dopo le 2 di notte o di cambiare la data del documento."]}}}
`
Cosa che non era mai successa prima in quella fascia oraria. Forse il down di ieri mattina c'entra qualcosa ?
In ogni caso vorrei sollecitare la riconsiderazione di questo inutile e dannoso limite !
grazie
AdM

**Comment by valmoz** [comment, marked as answer]
Buongiorno @Psicogest 
abbiamo rimosso il blocco in questione. Ci faccia sapere se va tutto bene.

Grazie
Mauro
